File: wire-android-automation/common/src/main/java/com/wearezeta/auto/common/calling2/v1/model/Instance.java
package com.wearezeta.auto.common.calling2.v1.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Instance {

	private final String id;
	private final InstanceStatus instanceStatus;

	private final String email;
	private final String password;
	private final Call currentCall;
	private final String backend;
	private final VersionedInstanceType instanceType;
	private final String screenshot;
	private final String name;
	private final long timeout;
	private final long created;
	private String message;

	@JsonCreator
	public Instance(@JsonProperty("id") String id,
			@JsonProperty("instanceStatus") InstanceStatus instanceStatus,
			@JsonProperty("currentCall") Call currentCall,
			@JsonProperty("email") String email,
			@JsonProperty("password") String password,
			@JsonProperty("backend") String backend,
			@JsonProperty("screenshot") String screenshot,
			@JsonProperty("instanceType") VersionedInstanceType instanceType,
			@JsonProperty("name") String name,
			@JsonProperty("timeout") long timeout,
			@JsonProperty("created") long created,
			@JsonProperty("message") String message) {
		this.id = id;
		this.instanceStatus = instanceStatus;
		this.currentCall = currentCall;
		this.email = email;
		this.password = password;
		this.backend = backend;
		this.screenshot = screenshot;
		this.instanceType = instanceType;
		this.name = name;
		this.timeout = timeout;
		this.created = created;
		this.message = message;
	}

	public String getId() {
		return id;
	}

	public InstanceStatus getStatus() {
		return instanceStatus;
	}

	public String getEmail() {
		return email;
	}

	public String getPassword() {
		return password;
	}

	public VersionedInstanceType getInstanceType() {
		return instanceType;
	}

	public long getTimeout() {
		return timeout;
	}

	public long getCreated() {
		return created;
	}

	public Call getCurrentCall() {
		return currentCall;
	}

	public String getLivePreview() {
		return screenshot;
	}

	public String getName() {
		return name;
	}

	@Override
	public String toString() {
		if (id == null && instanceStatus == null && message != null) return message;
		return "Instance{id=" + id + ", instanceStatus=" + instanceStatus
				+ ", email=" + email + ", password=" + password
				+ ", currentCall=" + currentCall + ", backend=" + backend
				+ ", instanceType=" + instanceType
				+ ", screenshot=<suppressed>, name=" + name + ", timeout="
				+ timeout + ", created=" + created + '}';
	}

}

File: tests/testsSupport/src/main/call/models/Instance.kt
/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 */
package call.models

import kotlinx.serialization.Serializable

@Serializable
data class Instance(
    val id: String? = null,
    val instanceStatus: InstanceStatus? = null,
    val currentCall: Call? = null,
    val email: String? = null,
    val password: String? = null,
    val backend: String? = null,
    val screenshot: String? = null,
    val instanceType: VersionedInstanceType? = null,
    val name: String? = null,
    val timeout: Long? = null,
    val created: Long? = null,
    val message: String? = null
) {
    override fun toString(): String {
        if (id == null && instanceStatus == null && message != null) return message
        return "Instance(id=$id, instanceStatus=$instanceStatus, email=$email, password=$password, " +
                "currentCall=$currentCall, backend=$backend, instanceType=$instanceType, " +
                "screenshot=<suppressed>, name=$name, timeout=$timeout, created=$created)"
    }
}
